### ISSUE TITLE
I got a BCM94360NG and the bluetooth is not working properly

### ISSUE THREAD

**Issue by shijiav** [Juan-VC/Hackintosh-macOS-Dell-G7-7588#60]
Do you have any idea what should be changed to get it work?
Currently the Bluetooth could only discover limited device, and disconnect after about 20 second.
I also pulled out a BCM94352Z from another old hackintosh laptop (DELL 7577) and plug it in.
And it can not discover any Bluetooth device (though Mac OS recognize it`s a 94352)

Im running 12.0.1

THX in advance

**Comment by Juan-VC-gh** [comment]
All bluetooth cards have issues on Monterey for now, this is due to the kernel extension that handle bluetooth. I also have a BCM94352Z and have noticed that incoming connections work but not outgoing so there is not much to do in that matter. Which card are you using that disconnects in about 20s?

**Comment by shijiav** [comment]
94360NG disconnect in about 20s, 94352Z behaves really strange, sometimes works and sometimes can not discover device at all.

**Comment by shijiav** [comment]
also both cards could not be recognized in hackintools, but mac os does recognize them.

**Comment by Juan-VC-gh** [comment]
But doesn't the 94360NG work even without drivers? About all the cards that need a driver there is not much to do than wait for the driver to get updated and fixed however I always thought that the 94360NG did not require drivers. If I am correct you may try removing them to see how it goes. My 94352Z behaves strange just as you report but there is not that much to do than wait for Acidanthera to fix bluetooth on Monterey

**Comment by 13thdemarch** [comment]
@shijiav BCM94360NG doesnt need any kexts. Remove all wifi/bluetooth kexts, then remove bluetooth module, reset network then reset nvram and try again
for NG card, you can use the usb mapping kext for broadcom card, because HS07 port is for internal usb bluetooth port

**Comment by ToCHONG** [comment]
> 94360NG disconnect in about 20s, 94352Z behaves really strange, sometimes works and sometimes can not discover device at all.

Did you solve that 20s disconnecting issue? I've just bought one and encounter that problem now... Thanks in advance

**Comment by Juan-VC-gh** [comment]
> @shijiav BCM94360NG doesnt need any kexts. Remove all wifi/bluetooth kexts, then remove bluetooth module, reset network then reset nvram and try again
> for NG card, you can use the usb mapping kext for broadcom card, because HS07 port is for internal usb bluetooth port

This is true that card should be run kextless, remove AirportBrcmFixup and BrcmPatchRAM kexts.
